sql/hive-thriftserver/src/test/scala/test/custom/listener/DummyQueryExecutionListener.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package test.custom.listener
+
+import org.apache.spark.sql.execution.QueryExecution
+import org.apache.spark.sql.util.QueryExecutionListener
+
+class DummyQueryExecutionListener extends QueryExecutionListener {
```

Comment:
There's a sort of weird Scala trick you can use to avoid these two source files. Just to this in the test suite source file:

```
package __root__.test.custom.listener {
  
  class DummyQueryExecutionListener { ...
    
  }
  
}
```

Then the test source is more self-contained, and you can attach the comment about the package name to the package declaration.

Reply:
Nice trick. I guess Spark respects package = directory most of times so place them to new directory intuitively. Given we seem to allow exceptions for such like this case, I'll apply the trick. Thanks!